app/features/settings/settings-actions.server.ts

```diff
@@ -19,57 +16,43 @@ import {
   settingsAccountSchema,
   settingsUserProfileSchema,
 } from './settings-client-schemas';
-import { withUsersOwnedOrganizations } from './settings-middleware.server';
+import { getUsersOwnedOrganizations } from './settings-helpers.server';
+
+export const settingsUserProfileAction = async ({
+  request,
+}: Pick<ActionFunctionArgs, 'request' | 'params'>) => {
+  const user = await requireOnboardedUserProfileExists(request);
+  const t = await i18next.getFixedT(request);
+
+  const data = await parseFormData(settingsUserProfileSchema, request);
```

Comment:
In a different PR, you refactored a function as an object wrapper for this function so it is unary. Now you use it normally with a an arity of two. Let's stick with one style please.

Also, all of these calls can be parallelized via `promiseHash`.